EasyClangComplete.sublime-settings

```diff
@@ -116,4 +116,20 @@
   // Controls if libclang will cache the results.
   // This works faster, but in rare cases can generate wrong completions.
   "use_libclang_caching": true,
+
+  // Templates to find source files for headers in case we use a
+  // compilation database: Such a DB does not contain the required
+  // compile flags for header files. In order to find a best matching
+  // source file instead, you can use templates. Such templates describe how
+  // to find (relative to the header file) a source file which we can
+  // use to get compile flags for. You can use UNIX style globbing patterns.
+  // In addition, you can use some pre-defined placeholders that are derived
+  // from the file we try to look up flags for:
+  // - basename: The base file name without the directory part.
+  // - stamp: Like "basename", but with the file name extension removed.
+  // - ext: The file name extension of the header file.
+  "header_to_source_mapping": [
```

Comment:
Can we just point to directories? It avoids educating the user on that a `stamp` is and reduces the number of variables that we need to provide. What do you think?

Reply:
I'd actually prefer being able to point to individual files. First, if we are in a mixed C/C++ project, that gives the user the chance to distinguish between both languages. And second, it allows you to have more fine grained control on the lookup strategy. For example, if you specify the following mapping:

```python
"header_to_source_mapping": [ "../src/{stamp}.cpp", "../src/*.*", ".*" ]
```

This would basically result in the following lookup strategy:

1. Check if in `../src` there is a file which we assume is the source file belonging to the current header file.
2. Otherwise, check if in `../src` there is at all a source file.
3. Otherwise, check the current directory.
4. *If everything else fails, use the known "merge all flags" strategy.*

However, if you think it makes (e.g. from maintenance point of view) more sense to simplify, I can change the implementation to point to directories. Alternatively, maybe even both could be supported (i.e. if one puts something like `../src/` with a trailing `/` into the mapping, we could look for any file in that directory - this is equivalent to `../src/*.*`).

Reply:
Ok, I am convinced :) 
This seems indeed a powerful tool and would be a pity to change, however, I would love to add your last option to it, so that we can just specify the folder. 

My logic is that if you bash users with too much settings they will never start using it in the first place. So I feel that the best would be to provide sane defaults that look simple to use while providing a full blown tutorial (ahem, issue #274) on how to use it in complex situations. How would you feel about it?

Reply:
I completely agree on that ;-) My implementation is - shame - mostly driven by my own needs. However, it definitely makes sense to keep things simple by default and let the user scale up if required. I will try to implement this as well.

Reply:
Cool. Thanks so much. Good implementations always come from personal needs :) I will play a role or a tyrant that wants to simplify things, so don't worry too much. And thanks again for doing this. This addition to the plugin looks really awesome :+1:

Reply:
Adding folders in the mapping table is now implemented in the latest version. Basically, this is done by "flattening" any entry that ends with a path delimiter into two entries,e.g. `../src/` becomes `../src/{stamp}.*` and `../src/*.*` - this is to prefer the likely source file belonging to a header to a possibly unrelated file.

In addition, the `CompilationDb` class now uses `{stamp}.*` and `*.*` as default patterns if they are not already part of the configuration; they are, however, added to the end of the pattern list so users can override the behavior by specifying other patterns in their configs.

Hope that somehow matches up with your expectations ;-)

Reply:
Yes, a 100% match. Thanks a lot! :)